code/modules/chemistry/Chemistry-Reagents.dm

```diff
@@ -160,6 +160,23 @@ datum
 							did_not_react = 0
 					if (ishuman(M))
 						var/mob/living/carbon/human/H = M
+
+						// Reset hair dye
+						var/had_dye = FALSE
```

Comment:
This will make hair dye wash out from stepping into a puddle of water or even blood with no shoes on, probably a little extreme anyway since water isn't known for removing hair dye. Hair dye removers are typically sulfur based according to wikipedia so maybe make sulfur wash off hair dye?